app/services/melanges/update_melange.rb

```diff
@@ -0,0 +1,26 @@
+module Melanges
+  class UpdateMelange < ApplicationService
+    def call
+      update_melange
```

Comment:
tutaj wyżej dodałbym 

<code> return false unless @melange.valid? </code>

metodę update_melange bym usunął i w call dał 

@melange.update!(@melange_params)

Transakcja na pojedynczym zapisie do bazy jest zbędna, bo Rails'y domyślnie i tak to robią.